beps/0012-metrics-service/README.md

```diff
@@ -0,0 +1,248 @@
+
+By providing a core metrics service:
+
+- **Plugin Authors** and the **Community** gain a straightforward way to address metrics instrumentation and can focus on business logic instead of needing to reimplement metrics plumbing.
+- **Backstage Admins** receive a reliable stream of metrics from the core system for monitoring, alerting, and troubleshooting.
+
+### Goals
+
+- Plugin-scoped metric namespacing
+- Consistent metrics patterns across all plugins
+- Aligned with OpenTelemetry industry standards
+- Provide a familiar interface as other core services
+
+The catalog and scaffolder plugins will be updated to use the new metrics service in the initial alpha release.
+
+### Non-Goals
+
+- Providing a way to configure the OpenTelemetry SDK. This is out of scope for this BEP.
+- Adding metrics to plugins missing existing metrics (outside of catalog and scaffolder)
+- Tracing and other telemetry concerns are out of scope for this BEP.
+- Refactoring the existing `LoggerService`. Future work to unify observability related concerns would be ideal, but not a goal.
+
+## Proposal
+
+Following similar patterns to other core services, create a new `RootMetricsService` responsible for root-level concerns and the creation of plugin-specific `MetricsService` instances.
+
+## Naming Conventions
+
+All Backstage metrics follow this hierarchical pattern:
+
+`backstage.{scope}.{scope_name}.{metric_name}`
+
+**Where:**
+
+- `backstage` is the root namespace for all Backstage metrics
+- `{scope}` is the system scope (either **plugin** or **framework**)
+- `{scope_name}` is the name of the plugin or framework service (e.g., `catalog`, `scaffolder`, `database`, `scheduler`)
+- `{metric_name}` is the hierarchical metric name as provided by the plugin author (e.g., `entities.processed.total`, `tasks.completed.total`)
+
+### Scope
+
+The `scope` represents where it belongs in the Backstage ecosystem.
+
+- `plugin` - A plugin-specific metric (e.g. `backstage.plugin.catalog.entities.count`)
+- `framework` - A metric provided by the framework (e.g. `backstage.framework.database.connections.active`)
+
+### Plugin-Scoped Metrics
+
+Pattern: `backstage.plugin.{pluginId}.{metric_name}`
+
+```yaml
+# Examples
+backstage.plugin.catalog.entities.processed.total
+backstage.plugin.scaffolder.tasks.completed.total
+backstage.plugin.techdocs.builds.active
+backstage.plugin.auth.sessions.active.total # todo: technically a core service and a backend plugin
+```
+
+### Framework-Scoped Metrics
+
+Pattern: `backstage.framework.{service}.{metric_name}`
+
+```yaml
+# Examples
+backstage.framework.database.connections.active
```

Comment:
We've used `core` in most of the places before. So thinking that:

`backstage.core.scheduler.tasks` might be a better fit?

Reply:
Maybe even just `core.scheduler`?

Reply:
We settled on `core` in the sig today.